pandas/_libs/pandas/type.h

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright (c) 2022-, PyData Development Team
+All rights reserved.
+
+Distributed under the terms of the BSD Simplified License.
+
+The full license is in the LICENSE file, distributed with this software.
+*/
+
+#ifndef PANDAS__LIBS_PANDAS_TYPE_H_
+#define PANDAS__LIBS_PANDAS_TYPE_H_
+#include "pyerrors.h"
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define PY_SSIZE_T_CLEAN
+#include <Python.h>
+#include <numpy/ndarrayobject.h>
+
+/*
+Cython equivalent of `isinstance(val, np.timedelta64)`
+
+Parameters
+----------
+val : object
+
+Returns
+-------
+is_timedelta64 : bool
+*/
+int is_timedelta64_object(PyObject *obj) {
+  return PyObject_TypeCheck(obj, &PyTimedeltaArrType_Type);
```

Comment:
Not sure the implementations ulltimately should be in this header file. Done for now just to emulate the existing `util.pxd` file, but would make the build system a bit easier with a dedicated .c implementation